scripts/publish-documentation/publish.sh

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env bash
+
+#
+# Copyright 2022, TeamDev. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Redistribution and use in source and/or binary forms, with or without
+# modification, must retain the above copyright notice and the following
+# disclaimer.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+PRIMARY_MARK="*"
+USAGE_TEMPLATE="Usage: publish.sh repositoryUrl='' tags='x,${PRIMARY_MARK}y,z' modules='x,y,z'"
+USAGE_EXAMPLE="Example: publish.sh repositoryUrl='https://github.com/SpineEventEngine/core-java.git' tags='v1.7.0,${PRIMARY_MARK}v1.8.0' modules='core,client'"
+
+# Check that exactly three parameters were provided.
+if [ "$#" -ne 3 ]; then
+    echo "$USAGE_TEMPLATE"
+    echo "$USAGE_EXAMPLE"
+    exit 22 # Invalid argument
+fi
+
+# Declare variables matching those passed to the script.
+for arg in "$@"
+do
+   key=$(echo "$arg" | cut -f1 -d=)
+
+   key_length=${#key}
+   value="${arg:key_length+1}"
+
+   declare "$key"="$value"
+done
+
+# Check that all necessary for the script parameters were set.
+if [ -z "$repositoryUrl" ] || [ -z "$tags" ] || [ -z "$modules" ]; then
+    echo "$USAGE_TEMPLATE"
+    echo "$USAGE_EXAMPLE"
+    exit 22 #Invalid argument
+fi
+
+mkdir "workspace" && cd "workspace" || exit 2 # Folder does not exist.
+git clone --branch="1.x-dev" "$repositoryUrl" "."
+
+log() {
+  echo "-----------------$1-----------------"
+}
+
+for tag in $(echo "$tags" | tr "," "\n")
+do
+  if [[ $tag == "$PRIMARY_MARK"* ]]; then
+    # Remove the leading $PRIMARY_MARK.
+    tag=${tag:1}
+    is_primary=true
+  else
+    is_primary=false
+  fi
+
+  log "Started working on the $tag tag."
```

Comment:
Could you please enclose the variable values into backticks? It's here and below. E.g. 
```
log "Started working on the `$tag` tag."
```
I am not sure if it works this way. But in any case, some formatting would feel nice in such cases.